### ISSUE TITLE
Can I run frankenphp with supervisor?

### ISSUE THREAD

**Issue by kilwir76** [dunglas/frankenphp#633]
### Describe you feature request

I'd like to be able to run daemon jobs via supervisor.

Is this possible or not? 

I've tried, but I can't run supervisor with frankenphp, even though it's installed in a Dockerfile.

**Comment by dunglas** [comment]
Yes, for instance Laravel Sail runs FrankenPHP using supervisord: https://laravel.com/docs/10.x/octane#frankenphp-via-laravel-sail

**Comment by kilwir76** [comment]
i need to run without laravel sail + octane. Its possible ?

**Comment by dunglas** [comment]
Definitely!

**Comment by sneycampos** [comment]
> i need to run without laravel sail + octane. Its possible ?

You can see the docs ([link](https://frankenphp.dev/docs/docker/#how-to-install-more-php-extensions)) to install supervisor. Something like this:

Dockerfile
```Dockerfile
FROM dunglas/frankenphp

RUN apt update && apt install supervisor

COPY supervisord.conf /etc/supervisor/conf.d/supervisord.conf

ENTRYPOINT ["/usr/bin/supervisord", "-c", "/etc/supervisor/conf.d/supervisord.conf"]
```

supervisord.conf
```conf
[program:octane]
command=php /app/artisan octane:frankenphp --host=127.0.0.1 --port=80 --admin-port=2019 --workers=16
```

**Comment by sneycampos** [comment]
@kilwir76 solved your problem? could you close the issue? :)

**Comment by kilwir76** [comment]
Thanks ! yes sorry for the wait

**Comment by kilwir76** [comment]
its a same fonctionnality on symfony ?

**Comment by sneycampos** [comment]
> its a same fonctionnality on symfony ?

Yes. you only need to set your entrypoint command in supervisord configuration. 

Supervisor is a process control system. It's not exclusive for laravel or anything. You can control any command with.

**Comment by thdebay** [comment]
What would be the supervisor command to run a Laravel app in the same way as the default Frankenphp image does? (without Octane)

**Comment by kitro** [comment]
Just wondring is there benefits of running frankenphp in supervisor than doing an entrypoint like this `ENTRYPOINT ["php", "artisan", "octane:frankenphp", "--caddyfile=/app/Caddyfile"]` with `restart: always` in docker service 🤔 .
I think the entrypoint is simple and achieve same need as running in supervisor.

**Comment by dunglas** [comment]
Supervisor is indeed useless if you're using Docker Compose.

**Comment by janklan** [comment]
> Supervisor is indeed useless if you're using Docker Compose.

Why useless? There are other things that can benefit from it, such as Messenger workers. Or does FrankenPHP somehow replace Supervisor role in that context?

**Comment by dunglas** [comment]
FrankenPHP doesn't, but Docker Compose does: https://stackoverflow.com/a/57564372

**Comment by withinboredom** [comment]
> Why useless?

Why run a supervisor on a supervisor?

**Comment by thdebay** [comment]
@kitro For me the benefit was mainly to handle all the processes required by my application in the same way (frankenphp, horizon, reverb...)

**Comment by sneycampos** [comment]
> [@kitro](https://github.com/kitro) For me the benefit was mainly to handle all the processes required by my application in the same way (frankenphp, horizon, reverb...)

But you can run each "process" as a service in docker compose, like this:

![Image](https://github.com/user-attachments/assets/757b0e6b-6b07-4b2e-8f42-32d749340569)

**Comment by janklan** [comment]
> > Why useless?
> 
> Why run a supervisor on a supervisor?

TL;DR I misunderstood @dunglas's comment. He's talking about it being useless to use Docker compose to start Supervisor that manages FrankenPHP. I agere with that, that's pointless. 

For context, I found this topic when I was trying to figure out how to run supervisor inside a dedicated FrankenPHP container where the user was not root. I was running into some permission issues (now solved), so I started searching, and found this.

We run supervisor next to the frankenphp process to manage additional jobs running next to frankenphp. Locally, we use Docker compose to start the FrankenPHP container. That's where my confusion with "Supervisor is indeed useless if you're using Docker Compose." started.

**Comment by sneycampos** [comment]
> [@sneycampos](https://github.com/sneycampos) I understand your point, thank you for taking the time to explain. I think you're right, your approach is better and I will look into it, maybe I will get rid of supervisor in my setup as well.

I used to run my apps with supervisor too, its ok, but since we are using docker-compose we really don't need supervisor

[another project](https://serversideup.net/open-source/docker-php/docs/laravel/laravel-task-scheduler) uses the same approach (queue, scheduler, horizon, etc)

**Comment by henderkes** [comment]
The rpm/deb packages have systemd integration. the apk packages have openrc integration. Supervisord is not planned, but you can write it yourself. launchd (macos/brew) and windows service integrations are planned and tracked separately. In docker, docker is the supervisor.

So I think there's no action left here.
